SECP-service/src/main/java/com/visucius/secp/auth/TokenAuthFilter.java

```diff
@@ -0,0 +1,79 @@
+package com.visucius.secp.auth;
+
+import io.dropwizard.auth.AuthFilter;
+import io.dropwizard.auth.AuthenticationException;
+import io.dropwizard.auth.Authenticator;
+import io.dropwizard.auth.basic.BasicCredentialAuthFilter;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.SecurityContext;
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Optional;
+
+@Priority(Priorities.AUTHENTICATION)
+public class TokenAuthFilter<P extends Principal> extends AuthFilter<String, P> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TokenAuthFilter.class);
+
+    @Override
+    public void filter(ContainerRequestContext containerRequestContext) throws IOException {
+        final String authToken = containerRequestContext.getHeaders().getFirst("token");
+
+        if (StringUtils.isBlank(authToken)) {
+            LOGGER.warn("Error decoding credentials");
+            throw new WebApplicationException(unauthorizedHandler.buildResponse(prefix, realm));
+        }
+
+        try {
+            final Optional<P> principal = authenticator.authenticate(authToken);
+            if (principal.isPresent()) {
+                containerRequestContext.setSecurityContext(new SecurityContext() {
```

Comment:
your creating overrides on SecurityContext object based on a conditional? Why not create a new class and use that instead?

Reply:
Yeah I understand what you mean. It would be cleaner that way. Just that if i create a class, i'll not be able to access the right authorizer, principal and containerRequestContext. take a look at https://dzone.com/articles/custom-security-context-injax-rs